src/Features/CSharp/Portable/Snippets/CSharpSnippetService.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets
+{
+    [ExportLanguageService(typeof(ISnippetService), LanguageNames.CSharp), Shared]
+    internal class CSharpSnippetService : ISnippetService
+    {
+        private readonly IEnumerable<Lazy<ISnippetProvider>> _snippetProvider;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpSnippetService([ImportMany] IEnumerable<Lazy<ISnippetProvider, LanguageMetadata>> snippetProvider)
+        {
+            _snippetProvider = snippetProvider;
+        }
+
+        public ISnippetProvider GetSnippetProvider(SnippetData data)
+        {
+            foreach (var provider in _snippetProvider)
+            {
+                if (data.DisplayName == provider.Value.GetSnippetText())
+                {
+                    return provider.Value;
```

Comment:
consider switching to a dictionary so this doesn't cost more and more as we add snippets.